src/service/alerts/scheduler.rs

```diff
@@ -1045,50 +1017,99 @@ async fn handle_derived_stream_triggers(
                 }
             }
 
-            match ingestion_error_msg {
-                None => db::scheduler::update_trigger(new_trigger).await?,
-                Some(err) => {
-                    if trigger.retries + 1 >= max_retries {
-                        // It has been tried the maximum time, just update the
-                        // next_run_at to the next expected trigger time
-                        log::debug!(
-                            "This DerivedStream trigger: {}/{} has reached maximum retries",
-                            &new_trigger.org,
-                            &new_trigger.module_key
-                        );
-                        trigger_data_stream.next_run_at = new_trigger.next_run_at;
-                        db::scheduler::update_trigger(new_trigger).await?;
-                    } else {
-                        // Otherwise update its status only
-                        db::scheduler::update_status(
-                            &new_trigger.org,
-                            new_trigger.module,
-                            &new_trigger.module_key,
-                            db::scheduler::TriggerStatus::Waiting,
-                            trigger.retries + 1,
-                        )
-                        .await?;
-                    }
+            if let Some(err) = ingestion_error_msg {
+                if trigger.retries + 1 >= max_retries {
+                    // It has been tried the maximum time, just update the next_run_at to the
+                    // next expected trigger time
+                    log::debug!(
+                        "This DerivedStream trigger: {}/{} has reached maximum retires",
+                        &new_trigger.org,
+                        &new_trigger.module_key
+                    );
+                    trigger_data_stream.next_run_at = new_trigger.next_run_at;
+                } else {
+                    // Otherwise, update its status only
+                    db::scheduler::update_status(
+                        &new_trigger.org,
+                        new_trigger.module.clone(),
+                        &new_trigger.module_key,
+                        db::scheduler::TriggerStatus::Waiting,
+                        trigger.retries + 1,
+                    )
+                    .await?;
                     trigger_data_stream.status = TriggerDataStatus::Failed;
-                    trigger_data_stream.error = Some(format!(
-                        "error saving enrichment table for DerivedStream: {err}"
-                    ));
+                    trigger_data_stream.error = Some(err);
                 }
+                // set end to now to exit the loop
+                end = now;
+            }
```

Comment:
This was intentional. Skipped period in this execution will be covered in the next execution since the current end_time is set to next run's start_time